test/recipes/checkhandshake.pm

```diff
@@ -0,0 +1,128 @@
+#! /usr/bin/env perl
```

Comment:
I'm not sure I like this location for a package.  We're spreading out a little too much.  OpenSSL::Test in `test/testlib`, TLSProxy in `util` and now this in `test/recipes`.  Why?

If it were me, I'd move this to `test/testlib` and simply `use checkhandshake;`, that'll save you the while `use lib <yada yada>` stuff as well, since `test/run_tests.pl` already sets up `test/testlib` as a package directory.